### PR TITLE
Overhauled primitives to be normal editors

### PR DESCRIPTION
**Pull request by johnsonr**
This aligns editor primitives with normal editors, via field injection. This means creating an independent instance in each case, hence we now need a creator function, not instances.

### CODE REVIEW

File: .atomist/editing/Primitives.ts [hunk 1]
```diff
@@ -1,20 +1,67 @@
...
 
+@Editor(UpdateFileEditor.Name, "Update a file")
+export class UpdateFileEditor extends FileContentEditor implements EditorPrimitive {
+
+    public static Name = "UpdateFile";
+
+    public id = UpdateFileEditor.Name;
+
+    public edit(project: Project): void {
+        const f = project.findFile(this.path);
```
Comment (johnsonr) [inline .atomist/editing/Primitives.ts]: Will add a null check here and push

File: .atomist/cascade/FeatureBroadcaster.ts [hunk 2]
```diff
@@ -9,9 +12,15 @@ import { EditProject } from "@atomist/rug/operations/ProjectEditor";
  */
 export class FeatureBroadcaster implements ScriptGenerator<Impact> {
 
+    /**
+     * Is the feature installable? That is, can we add it to existing projects?
+     * @type {boolean}
+     */
+    public installable: boolean = !!this.installer;
```
Comment (ddgenome) [inline .atomist/cascade/FeatureBroadcaster.ts]: Is this a common idiom to force something into a `boolean`?

File: .atomist/editing/Primitives.ts [hunk 3]
```diff
@@ -1,20 +1,69 @@
...
 
    public static Name = "AddFile";
 
-   public id = AddFilePrimitive.Name;
+   public id = AddFileEditor.Name;
 
-   public edit(project: Project, payload: any): void {
-     project.addExecutableFile(payload.path, payload.content);
+   public edit(project: Project): void {
+       // TODO why is this executable?
```
Comment (ddgenome) [inline .atomist/editing/Primitives.ts]: Seems like there is no need for it to be.

File: .atomist/editing/Primitives.ts [hunk 4]
```diff
@@ -1,20 +1,69 @@
...
+        displayName: "Path of file",
+        description: "Path of file",
+        pattern: "^[\/0-9a-zA-Z_\$\-]+$",
+        validInput: "Valid file path (Unix style)",
+        required: true,
+    })
+    public path: string;
+}
+
+abstract class FileContentEditor extends FileEditor {
```
Comment (ddgenome) [inline .atomist/editing/Primitives.ts]: What is the intent behind the FileEditor > FileContentEditor > AddFileEditor hierarchy? Only the last implements the edit method.

File: .atomist/cascade/FeatureBroadcaster.ts (hunk 2 above)
Comment (johnsonr) [inline .atomist/cascade/FeatureBroadcaster.ts]: Apparently

File: .atomist/editing/Primitives.ts (hunk 4 above)
Comment (johnsonr) [inline .atomist/editing/Primitives.ts]: Inheritance of parameters, as apparently spreads don't work

File: .atomist/editing/Primitives.ts (hunk 3 above)
Comment (johnsonr) [inline .atomist/editing/Primitives.ts]: Agree I'll get rid of it next time I touch that file

**Pull request merged**